--- rake/loaders/makefile.rb.orig    2008-07-22 19:01:11.000000000 +0300
+++ rake/loaders/makefile.rb    2008-07-22 19:27:03.000000000 +0300
@@ -9,16 +9,10 @@
   def load(fn)
     buffer = ''
     open(fn) do |mf|
-        mf.each do |line|
-          next if line =~ /^\s*#/
-          buffer << line
-          if buffer =~ /\\$/
-            buffer.sub!(/\\\n/, ' ')
-            state = :append
-          else
-            process_line(buffer)
-            buffer = ''
-          end
+        lines = mf.read
+    lines.gsub!(/\\\n/, ' ')
+        lines.split("\n").each do |line|
+          process_line line
       end
     end
     process_line(buffer) if buffer != ''
